### ISSUE TITLE
Brainstorm ideas for improved community engagement

### ISSUE THREAD

**Issue by littledan** [littledan/js-outreach-groups#4]
Even with these calls, there still is more room for improvement in how TC39 works with the community. If you have any suggestions, I would be happy to hear them.

**Comment by EisenbergEffect** [comment]
Just tossing ideas out as they pop in my head...

* A weekly or monthly "office hours" or "standup" video call that allows people to tune in and stay up to date and post questions on twitter.
* Some sort of official written communication channel that can be subscribed to, such as a tc39 blog.
* More "average" web developer-centric documentation of use cases during the specification process and more code examples as part of the use cases.
* A regular sync with major framework/library leaders.
* Official meeting transcripts or recordings of tc39 member meetings (maybe these exists and I just don't know where they are...which would signal a different issue around how information is shared.)
* Locate all proposals under the tc39 org or even consider locating all proposals within a single repo, perhaps using a monorepo structure.
* Standardize on naming of proposal repos and/or of issue naming and tagging.
* Represent tc39 working group status consistently in github tags across repos/issues.

I realize all this is a lot of work, but it is valuable. I confess that I know precious little about whether anyone is actually working on tc39 stuff full-time or if it's all just part-time work, sponsored by member companies. It could be a tremendous help to have tc39 directly fund a Technical Program Manager fulltime to organize and manage this type of thing for the language.

(I should note that I'd give the exact same advice to W3C and WHATWG who both don't do a great job of any of this either.)

**Comment by hax** [comment]
@EisenbergEffect 

> Official meeting transcripts or recordings of tc39 member meetings 

https://github.com/rwaldron/tc39-notes/

> a single repo, perhaps using a monorepo structure.

If it's a single repo, a controversial issue (like class fields, or recently globalThis naming issue) could flush the issue list. 

But I think a centralized repo which cover all status and simple introductions of all proposals could be helpful. Maybe current https://github.com/tc39/proposals is what you want, though you still need to go to specific repo to understand what a proposal is. 

> W3C and WHATWG who both don't do a great job of any of this either.

Standardizing is difficult. 😝

But I don't think W3C and WHATWG did anything worse than current ECMAScript class fields proposal, as my 20 years experiences of web developing and tracking of web standards.

**Comment by littledan** [comment]
@EisenbergEffect These are all great ideas, and we definitely have work to do in these areas. Some comments:

> A weekly or monthly "office hours" or "standup" video call that allows people to tune in and stay up to date and post questions on twitter.

@codehag is working on setting up a regular, open office hours call. She can provide more details.

A lot of us in TC39 are already on Twitter, chatting with people about stuff. We've talked about using an official Twitter account, but as TC39 consists of many people with different points of view, it could be difficult to figure out how to manage this sort of central messaging.

> Some sort of official written communication channel that can be subscribed to, such as a tc39 blog.

That's a good idea. @johannes-weber has suggested something along these lines as well, but we haven't gotten anything off the ground yet. Would anyone be interested in collaborating here?

> More "average" web developer-centric documentation of use cases during the specification process and more code examples as part of the use cases.

This is supposed to be part of the README files in TC39 proposal repositories. To the extent that it's not, IMO we should fix it. It's a lot of work; one way people can get involved is making issues PRs to add these things, but ultimately proposal champions should be responsible for making sure they are there.

> A regular sync with major framework/library leaders.

How do you think our framework sync is going?

> Official meeting transcripts or recordings of tc39 member meetings (maybe these exists and I just don't know where they are...which would signal a different issue around how information is shared.)

These are at https://github.com/tc39/tc39-notes/ . We're trying to make them more discoverable with a draft new website, at https://tc39.github.io/beta/ .

> Locate all proposals under the tc39 org or even consider locating all proposals within a single repo, perhaps using a monorepo structure.

Generally, we try to keep proposals that are being discussed in TC39 in the tc39 org. Having separate repos lets each repo have its own permissions set up, which is pretty useful. Do you see anything that we should change?

> Standardize on naming of proposal repos and/or of issue naming and tagging.

Issue tagging, that's something we don't have shared best practices on. Good idea.

For naming of repos, we use `tc39/proposal-xyz`. Is this standard enough?

> Represent tc39 working group status consistently in github tags across repos/issues.

Yes, this is important and I can see that we're not doing a great job here. It's hard, since there's already a lot of documentation in presentations, meeting notes, and issues, but we need to present a summary of the current status that goes beyond the up-to-date explainer.

I'm not sure how exactly we should present the current status. @chicoxyzzy has done a great job linking to meeting presentations and minutes in the class fields repo; do you think that helps?

For now, please ask questions about the status in issues on the repository, or in the outreach group meetings.

> I realize all this is a lot of work, but it is valuable. I confess that I know precious little about whether anyone is actually working on tc39 stuff full-time or if it's all just part-time work, sponsored by member companies. It could be a tremendous help to have tc39 directly fund a Technical Program Manager fulltime to organize and manage this type of thing for the language.

AFAIK I'm the only one working in TC39 full-time. I work for Igalia, which is a consultancy with clients. Most of my TC39 time is paid by Bloomberg; I share an interest with Bloomberg in pushing forward several technical proposals, but this sort of outreach stuff is extra on top. The new chair group (@codehag, @bterlson and @akirose) each have some time each week allocated for TC39 management, and could get involved in this sort of way as well.

I agree that it'd be great to have shared funding for a TPM for TC39 work. It's hard, though, since everyone comes with their own perspective. At the same time, if the TPM is held to a 100% "neutral" role (aside from that being a sort of fallacy and never really possible), they might be less effective in an environment where really strong opinions need to be worked through to make progress.

> (I should note that I'd give the exact same advice to W3C and WHATWG who both don't do a great job of any of this either.)

About the outreach groups specifically: I was thinking, if champions of W3C or WHATWG specs and the participants in an outreach group share an interest, we could discuss the topics in that group, inviting the champions; we don't need to limit things to TC39 topics.

**Comment by EisenbergEffect** [comment]
* For the meeting notes, it would be good to move those to the tc39 GitHub org and have them linked from a central readme there.
* Glad to hear that open office hours are being worked on.
* Framework sync from before was great. Let's just plan to continue on some sort of cadence.
* For repo location, I noticed that some were in other orgs or personal accounts, which is why I mention that. `proposal-` is fine as a naming convention. It's easy to find on a GitHub search that way if they are all under a single org.
* Super happy to hear about the new web site. That will go a long way. Also, please give me a way to subscribe to topics so I can know when things are happening :) I personally don't need or want to be involved in everything but I need a way to see high-level activity and become aware of things as fast as possible. If I see something that affects my open source community I want to be able to jump in right away.

**Comment by littledan** [comment]
Please refer to the notes repo within the tc39 org. If you see a proposal repo that is only in a personal repository, you can consider it very early, and focus on following things in the tc39 org.

We're having the framework sync monthly, and just cancelled the one December meeting because of winter holidays. Is it showing up in your calendar?

I will raise the idea of a resource to subscribe to with my colleagues in tc39. For now, the three repos many people watch to keep up with things are
- tc39/proposals, to see when proposals move through stages
- tc39/agendas, to see what we will discuss in meetings, and find links to presentations
- tc39/tc39-notes, to read committee minutes when they come out
We hope to make this more clear on the website.

**Comment by littledan** [comment]
@EisenbergEffect Yes, there was, sorry about this. I'll re-send you the invite for January.

**Comment by Regaddi** [comment]
Because it hasn't been mentioned yet:

I know there is the tc39 IRC channel. As far as I know not too many developers use IRC for their daily communication, but Slack, Spectrum, Gitter or another chat client at work.
A lot of open source projects use such chat platforms to give their users the ability to raise questions without polluting the issues area in the repositories.

However I can imagine, that using one of these services could potentially reduce the hurdles to get in contact directly with the tc39 people.
